src/basic/fs-util.c

```diff
@@ -1028,3 +1029,44 @@ int parse_cifs_service(
 
         return 0;
 }
+
+char *resolve_auto_dir(const char *path) {
```

Comment:
please propagate any errors we see via the return value. i.e. we typically prefer that the caller ignores errors when that's a good idea, not the callee.

Reply:
Changed the signature and logic for this